tokio-fs/tests/dir.rs

```diff
@@ -0,0 +1,89 @@
+extern crate futures;
+extern crate tempdir;
+extern crate tokio_fs;
+extern crate tokio_threadpool;
+
+use futures::sync::oneshot;
+use futures::{Future, Stream};
+use std::fs;
+use std::io;
+use std::sync::{Arc, Mutex};
+use tempdir::TempDir;
+use tokio_fs::*;
+use tokio_threadpool::Builder;
+
+fn run_in_pool<F>(f: F)
+where
+    F: Future<Item = (), Error = std::io::Error> + Send + 'static,
+{
+    let pool = Builder::new().pool_size(1).build();
+    let (tx, rx) = oneshot::channel::<()>();
+    pool.spawn(f.then(|_| tx.send(())));
+    rx.wait().unwrap()
+}
+
+#[test]
+fn create() -> io::Result<()> {
```

Comment:
Can you change these to not have a return type?  The tests on older versions of Rust don't seem to like it.

Or did you specifically make them `Result` because your compiler was complaining?

Reply:
Yeah, that's true.  I think a lot of tests don't use `Result<T, E>` now so we should make the tests pass and maybe we do another PR, if you want or anyone could do it, to upgrade the tests across tokio when we finally don't test against 1.26.